core/src/main/scala/org/apache/spark/deploy/rest/RestSubmissionClient.scala

```diff
@@ -184,10 +190,34 @@ private[spark] class RestSubmissionClient(master: String) extends Logging {
     message
   }
 
+  private class AllTrustManager extends X509TrustManager {
+    def getAcceptedIssuers: Array[X509Certificate] = null
+    def checkClientTrusted(certs: Array[X509Certificate], authType: String): Unit = {}
+    def checkServerTrusted(certs: Array[X509Certificate], authType: String): Unit = {}
+  }
+
+  val trustAllCerts = Array[TrustManager](new AllTrustManager)
+  val trustAllContext = SSLContext.getInstance("SSL")
+  trustAllContext.init(null, trustAllCerts, new java.security.SecureRandom())
+
+  private def urlConnection(url: URL): HttpURLConnection = {
+    val con = url.openConnection().asInstanceOf[HttpURLConnection]
+    if (url.getProtocol == "https") {
+      if (System.getProperty("spark.ssl.insecureSkipTLSVerify", "false") == "true") {
+        val scon = con.asInstanceOf[HttpsURLConnection]
```

Comment:
Can you add a LOG(INFO) here saying we're setting up insecure verification for SSL?